### ISSUE TITLE
Add driver support for passing in AuthTokenManager for latest neo4j driver

### ISSUE THREAD

**Issue by er1c** [neotypes/neotypes#762]
When creating a neotypes driver, I'd like to specify creation using the AuthTokenManager introduced in (around? `5.13`)

Ref: https://neo4j.com/docs/api/java-driver/current/org.neo4j.driver/org/neo4j/driver/AuthTokenManagers.html

https://github.com/neotypes/neotypes/blob/85312e07703d923839e8ef1b57f3722256e12762/core/src/main/scala/neotypes/GraphDatabase.scala#L58

https://neo4j.com/docs/api/java-driver/current/org.neo4j.driver/org/neo4j/driver/GraphDatabase.html

<img width="1327" alt="image" src="https://github.com/neotypes/neotypes/assets/64685/75e9a6a8-aa16-49bf-a201-15f8a61e4901">

**Comment by er1c** [comment]
FWIW, here's some java code that makes it work with AWS Neptune...was looking into porting it to scala

```java
package org.example;

import com.amazonaws.DefaultRequest;
import com.amazonaws.Request;
import com.amazonaws.auth.AWS4Signer;
import com.amazonaws.auth.AWSCredentialsProvider;
import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
import com.amazonaws.http.HttpMethodName;
import com.google.gson.Gson;
import org.neo4j.driver.AuthToken;
import org.neo4j.driver.AuthTokens;
import org.neo4j.driver.Config;
import org.neo4j.driver.Driver;
import org.neo4j.driver.GraphDatabase;
import org.neo4j.driver.Query;
import org.neo4j.driver.AuthTokenManagers;

import java.net.URI;
import java.time.ZonedDateTime;
import java.util.HashMap;
import java.util.Map;

import static com.amazonaws.auth.internal.SignerConstants.AUTHORIZATION;
import static com.amazonaws.auth.internal.SignerConstants.HOST;
import static com.amazonaws.auth.internal.SignerConstants.X_AMZ_DATE;
import static com.amazonaws.auth.internal.SignerConstants.X_AMZ_SECURITY_TOKEN;
import static org.neo4j.driver.Values.parameters;

/**
 * Playground example derived from Neo4j's HelloWorldExample sample
 */
public class Neo4jv5ToNeptuneExample implements AutoCloseable {
    private final Driver driver;
    private static final AWSCredentialsProvider AWS_CREDENTIALS_PROVIDER = new DefaultAWSCredentialsProviderChain();
    static final String SERVICE_NAME = "neptune-db";
    static final String HTTP_METHOD_HDR = "HttpMethod";
    static final String DUMMY_USERNAME = "username";
    static final long SECONDS_TO_EXPIRE = 300; // 5 minutes
    private static final Gson GSON = new Gson();

    public Neo4jv5ToNeptuneExample(String clusterUrl, String awsRegion) {

        // the auth token manager is what will refresh the IAM token based on timestamp set by expiringAt
        var authManager = AuthTokenManagers.bearer(() -> {
            // obtain the signed token
            var token = createAuthToken(clusterUrl, awsRegion);
            // set the refreshing expiry to be slightly earlier
            return token.expiringAt(ZonedDateTime.now().plusSeconds(SECONDS_TO_EXPIRE - 10).toInstant().toEpochMilli());
        });

        driver = GraphDatabase.driver(
                clusterUrl,
                authManager,
                Config.builder().withEncryption()
                        .withTrustStrategy(Config.TrustStrategy.trustSystemCertificates())
                        .build());
    }

    @Override
    public void close() throws RuntimeException {
        driver.close();
    }

    /**
     * Function to generate an AuthToken using IAM authentication.
     *
     * @param url    URL to point at.
     * @param region Region to use.
     * @return AuthToken for IAM authentication.
     */
    public static AuthToken createAuthToken(final String url, final String region) {
        final Request<Void> request = new DefaultRequest<>(SERVICE_NAME);
        request.setHttpMethod(HttpMethodName.GET);
        request.setEndpoint(URI.create(url));
        request.setResourcePath("/opencypher");

        final AWS4Signer signer = new AWS4Signer();
        signer.setRegionName(region);
        signer.setServiceName(request.getServiceName());
        signer.sign(request, AWS_CREDENTIALS_PROVIDER.getCredentials());

        return AuthTokens.basic(DUMMY_USERNAME, getAuthInfoJson(request));
    }

    /**
     * Helper function for AuthToken using IAM authentication.
     */
    private static String getAuthInfoJson(final Request<Void> request) {
        final Map<String, Object> obj = new HashMap<>();
        obj.put(AUTHORIZATION, request.getHeaders().get(AUTHORIZATION));
        obj.put(HTTP_METHOD_HDR, request.getHttpMethod());
        obj.put(X_AMZ_DATE, request.getHeaders().get(X_AMZ_DATE));
        obj.put(HOST, request.getHeaders().get(HOST));
        obj.put(X_AMZ_SECURITY_TOKEN, request.getHeaders().get(X_AMZ_SECURITY_TOKEN));

        return GSON.toJson(obj);
    }

    /**
     * Playground function.
     */
    public void printGreeting(final String message) {
        try (var session = driver.session()) {
            var greeting = session.executeWrite(tx -> {
                var query = new Query("CREATE (a:Greeting) SET a.message = $message RETURN a.message + ', from node ' + id(a)", parameters("message", message));
                var result = tx.run(query);
                return result.single().get(0).asString();
            });
            System.out.println(greeting);
        }
    }

    public static void main(String... args) {
        try (var greeter = new Neo4jv5ToNeptuneExample("bolt://cluster-url:8182", "us-east-2")) {
            greeter.printGreeting("hello, world");
        }
    }
}
```

**Comment by BalmungSan** [comment]
Fixed by #763 
I will make a release in a moment.

**Comment by BalmungSan** [comment]
`v1.1.0` was just released and should already be accessible
